compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -226,8 +227,14 @@ impl<'tcx> InferCtxtInner<'tcx> {
     #[inline]
     pub fn unwrap_region_constraints(&mut self) -> RegionConstraintCollector<'_, 'tcx> {
         self.region_constraint_storage
-            .as_mut()
-            .expect("region constraints already solved")
+            // Replace region constraints if they've been taken.
+            // This should only happen on an error path, so delay a bug.
```

Comment:
```suggestion
            // This should only happen on an error path, so delay a bug.
            //
            // This is only allowed so that one can use the existing inference
            // context for diagnostics.
```